src/Common/src/Certificates/ConfigureCertificateOptions.cs

```diff
@@ -42,16 +47,29 @@ public void Configure(string? name, CertificateOptions options)
 
         string? privateKeyFilePath = _configuration.GetValue<string>(GetConfigurationKey(name, "PrivateKeyFilePath"));
 
-        options.Certificate = privateKeyFilePath != null && File.Exists(privateKeyFilePath)
-            ? X509Certificate2.CreateFromPemFile(certificateFilePath, privateKeyFilePath)
-            : new X509Certificate2(certificateFilePath);
+        try
+        {
+            options.Certificate = privateKeyFilePath != null && File.Exists(privateKeyFilePath)
+                ? X509Certificate2.CreateFromPemFile(certificateFilePath, privateKeyFilePath)
+                : new X509Certificate2(certificateFilePath);
 
-        X509Certificate2[] certificateChain = CertificateRegex.Matches(File.ReadAllText(certificateFilePath))
-            .Select(x => new X509Certificate2(Encoding.ASCII.GetBytes(x.Value))).ToArray();
+            X509Certificate2[] certificateChain = CertificateRegex.Matches(File.ReadAllText(certificateFilePath))
+                .Select(x => new X509Certificate2(Encoding.ASCII.GetBytes(x.Value))).ToArray();
 
-        foreach (X509Certificate2 issuer in certificateChain.Skip(1))
+            foreach (X509Certificate2 issuer in certificateChain.Skip(1))
+            {
+                options.IssuerChain.Add(issuer);
+            }
+        }
+        catch (IOException ex)
+        {
+            _logger.LogDebug(ex, "IOException while loading certificate for '{CertificateName}' from '{Path}'. Will retry on next reload.", name,
```

Comment:
Can you point me to a place where this actually happens? I've added the following code to find out, but all tests remain green. I scanned for failing pipelines in AzDO, but didn't find any.

```c#
if (exception is IOException)
{
    throw null;
}
```